src/ng/compile.js

```diff
@@ -1877,6 +1877,36 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
         return value || null;
       }
 
+      function setupControllers($element, attrs, transcludeFn, controllerDirectives, isolateScope, scope) {
+        var elementControllers = createMap();
+        for (var controllerKey in controllerDirectives) {
```

Comment:
Personally I often prefer `for ... in` over `forEach`. That's all `forEach` does anyway.

Could use `hasOwnProperty` instead of `createMap` but that seems to be slower in many cases.  Although that would actually make it more like previously (because that's what `forEach` does) which would reduce the changes in this commit.

The other option is `Object.keys() + for (0 to len)`. Or maybe `controllerDirectives` could actually be an array? Looks like the only place `controllerDirectives` is used other then looping over the values is for the `assertNoDuplicate` check...